README
This archive contains the distribution Dist-Zilla-PluginBundle-MSCHOUT,
version 0.39:

  Use L<Dist::Zilla> like MSCHOUT does

This software is copyright (c) 2023 by Michael Schout.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v6.030.


Changes
Revision history for Dist::Zilla::PluginBundle::MSCHOUT

{{$NEXT}}

0.40  2023-08-18
    - Disable twitter support. RIP Twitter.

0.39  2023-08-18
    - Remove support for Travis CI

0.38  2018-06-04
    - Add Pod::Weaver::Plugin::SingleEncoding

0.37  2018-02-07
    - switch bugtracker from rt.cpan.org to github issues

0.36  2017-12-24
    - add Dist::Zilla::Plugin::MetaProvides::Package plugin

0.35  2017-08-03
    - remove ArchiveRelease

0.34  2017-07-31
    - add AuthorSignatureTest

0.33  2017-07-31
    - Git::Check: allow .travis.yml to be dirty

0.32  2017-07-29
    - add MinimumPerl plugin
    - enable Dist::Zilla::PluginBundle::PluginRemover plugin
    - make signature test an AUTHOR test and use Test::Signature
    - use Moose attibutes for options
    - enable ::PluginBundle::Config::Slicer role so plugin options can be
      easily set
    - enable InsertCopyright plugin
    - enable Prereqs::AuthorDeps plugin

0.31  2016-08-30
    - Configure Git::Push to also push build/releases if it exists on origin
    - signature test: skip generated files

0.30  2016-08-30
    - Mention use_travis and use_twitter options in POD docs
    - add MetaJSON plugin

0.29  2016-07-04
    - 0.28 accidentally completely removed Git::Push. Fixed.

0.28  2016-07-04
    - back out build/releases push config - this auto pushes anayway if remote
      branch exists.

0.27  2016-07-04
    - add use_travis option to enable the TravisYML plugin
    - configure Git::Push to push build/releases as well as master branch

0.26  2016-07-02
    - fix typo causing breakage if use_twitter was set (oops)

0.25  2016-07-02
    - enable Twitter plugin if 'use_twitter' is set
    - use RemovePrereqs to remove Module::Signature from prereqs since its
      optional
    - add missing dependency CheckPrereqsIndexed

0.24  2016-06-30
    - PodWeaver: allow SOURCE and AUTHOR(S) sections to be overridden.

0.23  2016-06-27
    - switch from deprecated BumpVersionFromGit to Git::NextVersion.  This is a
      drop-in replacement.

0.22  2011-05-30
    - use CheckPrereqsIndexed plugin
    - add "task" option which replaces Pod::Weaver with Task::Weaver, and uses
      AutoVersion

0.21  2011-05-13
    - use Pod::Elemental::Transformer::List

0.20  2011-02-01
    - add Pod::Weaver::PluginBundle::MSCHOUT

0.19  2011-01-25
    - code cleanups.
    - doc fixes.
    - add release_branch configuration arg

0.18  2011-01-24
    - enable Git::CommitBuild, building releases on build/releases branch.
    - Git::Tag tags on build/releases branch now

0.17  2011-01-18
    - update to work with Dist::Zilla 4.x [(Auto)Prereq -> (Auto)Prereqs)]
    - require Dist::Zilla 4.102341

0.16  2010-05-03
    - add NextRelease back in - was missing since 0.13

0.15
    - add Git::CommitBuild plugin

0.14
    - if no_upload is true, remove ConfirmRelease

0.13  2010-04-14
    - simplify with Dist::Zilla::PluginBundle::Easy
    - require Dist::Zilla 2.101040

0.12  2010-04-09
    - add Pod::Weaver::Section::SourceGitHub, Pod::Weaver::Section::BugsRT
      to prereq's
    - use ArchiveRelease plugin
    - add BumpVersionFromGit plugin
    - use FakeRelease releaser
    - BumpVersionFromGit fix first_version

0.11  2010-03-29
    - turn off CPAN uploads, use ArchiveRelease releaser plugin

0.10  2010-03-29
    - initial release


lib/Pod/Weaver/PluginBundle/MSCHOUT.pm
#
# This file is part of Dist-Zilla-PluginBundle-MSCHOUT
#
# This software is copyright (c) 2023 by Michael Schout.
#
# This is free software; you can redistribute it and/or modify it under
# the same terms as the Perl 5 programming language system itself.
#

package Pod::Weaver::PluginBundle::MSCHOUT;
$Pod::Weaver::PluginBundle::MSCHOUT::VERSION = '0.39';
# ABSTRACT: Pod::Weaver configuration the way MSCHOUT does it

# Dependencies
use Pod::Weaver::Section::SourceGitHub;
use Pod::Weaver::Section::Bugs;

use Pod::Weaver::Config::Assembler;

use namespace::autoclean;

sub _exp { Pod::Weaver::Config::Assembler->expand_package($_[0]) }

sub mvp_bundle_config {
    return (
        [ '@Default/CorePrep', _exp('@CorePrep'), {} ],
        [ '@Default/Name',     _exp('Name'),      {} ],
        [ '@Default/Version',  _exp('Version'),   {} ],

        [ '@Default/prelude', _exp('Region'),  { region_name => 'prelude' } ],
        [ 'SYNOPSIS',         _exp('Generic'), {} ],
        [ 'DESCRIPTION',      _exp('Generic'), {} ],
        [ 'OVERVIEW',         _exp('Generic'), {} ],

        [ 'ATTRIBUTES', _exp('Collect'), { command => 'attr' } ],
        [ 'METHODS',    _exp('Collect'), { command => 'method' } ],
        [ 'FUNCTIONS',  _exp('Collect'), { command => 'func' } ],

        [ '@MSCHOUT/Leftovers', _exp('Leftovers'), {} ],

        # SOURCE section using GitHub.  If present in POD already, POD value
        # will be used.
        [ '@MSCHOUT/SourceGitHub', _exp('SourceGitHub'), {} ],
        [ 'AllowOverride/Source', _exp('AllowOverride'), {
                header_re      => '^SOURCE$',
                action         => 'replace',
                match_anywhere => 0
            }
        ],
        [ '@MSCHOUT/Bugs',       _exp('Bugs'),       {} ],

        [ '@MSCHOUT/postlude', _exp('Region'), { region_name => 'postlude' } ],

        # AUTHOR section. Will use POD section if present already
        [ '@MSCHOUT/Authors', _exp('Authors'), {} ],
        [ 'AllowOverride/Authors', _exp('AllowOverride'), {
                header_re      => '^AUTHORS?$',
                action         => 'replace',
                match_anywhere => 0
            }
        ],
        [ '@MSCHOUT/Legal',   _exp('Legal'),   {} ],
        [ '@MSCHOUT/List',    _exp('-Transformer'), { transformer => 'List' } ],
        [ '@MSCHOUT/SingleEncoding', _exp('-SingleEncoding'), {} ],
    );
}

1;

__END__

=pod

=encoding UTF-8

=head1 NAME

Pod::Weaver::PluginBundle::MSCHOUT - Pod::Weaver configuration the way MSCHOUT does it

=head1 VERSION

version 0.39

=head1 DESCRIPTION

This is the L<Pod::Weaver> config I use for building my documentation.  I use
it with L<Dist::Zilla::PluginBundle::MSCHOUT>

=for Pod::Coverage mvp_bundle_config

=head1 SOURCE

The development version is on github at L<https://https://github.com/mschout/dist-zilla-pluginbundle-mschout>
and may be cloned from L<git://https://github.com/mschout/dist-zilla-pluginbundle-mschout.git>

=head1 BUGS

Please report any bugs or feature requests on the bugtracker website
L<https://github.com/mschout/dist-zilla-pluginbundle-mschout/issues>

When submitting a bug or request, please include a test-file or a
patch to an existing test-file that illustrates the bug or desired
feature.

=head1 AUTHOR

Michael Schout <mschout@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2023 by Michael Schout.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.030.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Use L<Dist::Zilla> like MSCHOUT does",
  "AUTHOR" => "Michael Schout <mschout\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-PluginBundle-MSCHOUT",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::PluginBundle::MSCHOUT",
  "PREREQ_PM" => {
    "Dist::Zilla" => "4.102341",
    "Dist::Zilla::Plugin::AuthorSignatureTest" => 0,
    "Dist::Zilla::Plugin::AutoPrereqs" => 0,
    "Dist::Zilla::Plugin::AutoVersion" => 0,
    "Dist::Zilla::Plugin::CheckPrereqsIndexed" => 0,
    "Dist::Zilla::Plugin::FakeRelease" => 0,
    "Dist::Zilla::Plugin::Git::NextVersion" => 0,
    "Dist::Zilla::Plugin::GithubMeta" => 0,
    "Dist::Zilla::Plugin::InsertCopyright" => 0,
    "Dist::Zilla::Plugin::MetaJSON" => 0,
    "Dist::Zilla::Plugin::MetaProvides::Package" => 0,
    "Dist::Zilla::Plugin::MinimumPerl" => 0,
    "Dist::Zilla::Plugin::NextRelease" => 0,
    "Dist::Zilla::Plugin::PodWeaver" => 0,
    "Dist::Zilla::Plugin::Prereqs::AuthorDeps" => 0,
    "Dist::Zilla::Plugin::RemovePrereqs" => 0,
    "Dist::Zilla::Plugin::Signature" => 0,
    "Dist::Zilla::Plugin::TaskWeaver" => "0.093330",
    "Dist::Zilla::Plugin::Twitter" => 0,
    "Dist::Zilla::PluginBundle::Classic" => 0,
    "Dist::Zilla::PluginBundle::Filter" => 0,
    "Dist::Zilla::PluginBundle::Git" => "1.101230",
    "Dist::Zilla::Role::PluginBundle::Config::Slicer" => 0,
    "Dist::Zilla::Role::PluginBundle::Easy" => 0,
    "Dist::Zilla::Role::PluginBundle::PluginRemover" => 0,
    "Moose" => 0,
    "MooseX::AttributeShortcuts" => 0,
    "Pod::Elemental::Transformer::List" => 0,
    "Pod::Weaver::Config::Assembler" => 0,
    "Pod::Weaver::Plugin::SingleEncoding" => 0,
    "Pod::Weaver::Section::AllowOverride" => 0,
    "Pod::Weaver::Section::Bugs" => 0,
    "Pod::Weaver::Section::SourceGitHub" => 0,
    "namespace::autoclean" => "0.09"
  },
  "TEST_REQUIRES" => {
    "Test::More" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "VERSION" => "0.39",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Dist::Zilla" => "4.102341",
  "Dist::Zilla::Plugin::AuthorSignatureTest" => 0,
  "Dist::Zilla::Plugin::AutoPrereqs" => 0,
  "Dist::Zilla::Plugin::AutoVersion" => 0,
  "Dist::Zilla::Plugin::CheckPrereqsIndexed" => 0,
  "Dist::Zilla::Plugin::FakeRelease" => 0,
  "Dist::Zilla::Plugin::Git::NextVersion" => 0,
  "Dist::Zilla::Plugin::GithubMeta" => 0,
  "Dist::Zilla::Plugin::InsertCopyright" => 0,
  "Dist::Zilla::Plugin::MetaJSON" => 0,
  "Dist::Zilla::Plugin::MetaProvides::Package" => 0,
  "Dist::Zilla::Plugin::MinimumPerl" => 0,
  "Dist::Zilla::Plugin::NextRelease" => 0,
  "Dist::Zilla::Plugin::PodWeaver" => 0,
  "Dist::Zilla::Plugin::Prereqs::AuthorDeps" => 0,
  "Dist::Zilla::Plugin::RemovePrereqs" => 0,
  "Dist::Zilla::Plugin::Signature" => 0,
  "Dist::Zilla::Plugin::TaskWeaver" => "0.093330",
  "Dist::Zilla::Plugin::Twitter" => 0,
  "Dist::Zilla::PluginBundle::Classic" => 0,
  "Dist::Zilla::PluginBundle::Filter" => 0,
  "Dist::Zilla::PluginBundle::Git" => "1.101230",
  "Dist::Zilla::Role::PluginBundle::Config::Slicer" => 0,
  "Dist::Zilla::Role::PluginBundle::Easy" => 0,
  "Dist::Zilla::Role::PluginBundle::PluginRemover" => 0,
  "Moose" => 0,
  "MooseX::AttributeShortcuts" => 0,
  "Pod::Elemental::Transformer::List" => 0,
  "Pod::Weaver::Config::Assembler" => 0,
  "Pod::Weaver::Plugin::SingleEncoding" => 0,
  "Pod::Weaver::Section::AllowOverride" => 0,
  "Pod::Weaver::Section::Bugs" => 0,
  "Pod::Weaver::Section::SourceGitHub" => 0,
  "Test::More" => 0,
  "namespace::autoclean" => "0.09",
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
